### PR TITLE
test shadowing function from base 

### LINKED ISSUE #11723: Method not found when shadowing function from base with a local variable
I have the strange error

```
ERROR: LoadError: MethodError: `convert` has no method matching convert(::Type{SDE.MvLinPro}, ::Array{Float64,2}, ::Array{Float64,1}, ::Array{Float64,2})
This may have arisen from a call to the constructor SDE.MvLinPro(...),
since type constructors fall back to convert methods.
Closest candidates are:
  SDE.MvLinPro(::Array{Float64,2}, ::Array{Float64,1}, ::Array{Float64,2})
```

I don't really now where to start, but can this be a reasonable message at all? Julia knows and doesn't know about the inner constructor method

```
  SDE.MvLinPro(::Array{Float64,2}, ::Array{Float64,1}, ::Array{Float64,2})
```

### PR DESCRIPTION
**Pull request by KristofferC**
Closes #11723

### CODE REVIEW

File: test/misc.jl [hunk 1]
```diff
@@ -794,3 +794,21 @@ end # module
...
+    # required to exhibit bug
+    diag  = ones(5)
+
+    A = Tridiagonal(lower,diag,upper)
+
+    for i=1:10 # required to exhibit bug
+    end
+end"""
+
+cmd = `$(Base.julia_cmd()) -e $str`
```
Comment (tkelman) [inline test/misc.jl]: startup-file=no

**Comment by KristofferC** [comment]
Heh, this passes on 0.6 but on 0.7-dev we get:

```
WARNING: imported binding for diag overwritten in module Main
Unreachable reached at 0x7fa65d05366a

signal (4): Illegal instruction
while loading no file, in expression starting on line 0
macro expansion at ./REPL[0]:7 [inlined]
anonymous at ./<missing> (unknown line)
jl_call_fptr_internal at /home/kristoffer/julia/src/julia_internal.h:366 [inlined]
jl_call_method_internal at /home/kristoffer/julia/src/julia_internal.h:385 [inlined]
jl_toplevel_eval_flex at /home/kristoffer/julia/src/toplevel.c:594
jl_toplevel_eval_in at /home/kristoffer/julia/src/builtins.c:496
eval at ./boot.jl:236
jlcall_eval_18457 at /home/kristoffer/julia/usr/lib/julia/sys.so (unknown line)
jl_call_fptr_internal at /home/kristoffer/julia/src/julia_internal.h:366 [inlined]
jl_call_method_internal at /home/kristoffer/julia/src/julia_internal.h:385 [inlined]
jl_apply_generic at /home/kristoffer/julia/src/gf.c:1928
eval_user_input at ./repl/REPL.jl:67
jlcall_eval_user_input_20521 at /home/kristoffer/julia/usr/lib/julia/sys.so (unknown line)
jl_call_fptr_internal at /home/kristoffer/julia/src/julia_internal.h:366 [inlined]
jl_call_method_internal at /home/kristoffer/julia/src/julia_internal.h:385 [inlined]
jl_apply_generic at /home/kristoffer/julia/src/gf.c:1928
macro expansion at ./repl/REPL.jl:98 [inlined]
#1 at ./event.jl:73
unknown function (ip: 0x7fa65d04de1f)
jl_call_fptr_internal at /home/kristoffer/julia/src/julia_internal.h:366 [inlined]
jl_call_method_internal at /home/kristoffer/julia/src/julia_internal.h:385 [inlined]
jl_apply_generic at /home/kristoffer/julia/src/gf.c:1928
jl_apply at /home/kristoffer/julia/src/julia.h:1445 [inlined]
start_task at /home/kristoffer/julia/src/task.c:267
unknown function (ip: 0xffffffffffffffff)
Allocations: 1717317 (Pool: 1715911; Big: 1406); GC: 1
[1]    11070 illegal hardware instruction (core dumped)  ./julia
```

**Comment by KristofferC** [comment]
cc @vtjnash @JeffBezanson

**Comment by martinholters** [comment]
Bisected the crash to 753fbef68dd3f09fb0137d6a78606db098119099. As this does not crash if either `diag` is renamed or the whole thing is wrapped in a function, my guess is that `Triangular(lower, Base.diag, upper)` is inferred, which is correctly determined as throwing a `MethodError`, but then `Triangular(lower, diag, upper)` is invoked, which does not throw, hence `Unreachable reached`. Sounds like  #11723 has never really been fixed. Especially for https://github.com/JuliaLang/julia/issues/11723#issuecomment-112471192 I still get the same output.

**Comment by martinholters** [comment]
What's the status here?

File: test/misc.jl [hunk 2]
```diff
@@ -794,3 +794,21 @@ end # module
...
+str = """
+if true # required to exhibit bug
+    upper = ones(4)
+    lower = ones(4)
+
+    # WARNING: imported binding overwritten
+    # required to exhibit bug
+    diag  = ones(5)
+
+    A = Tridiagonal(lower,diag,upper)
```
Comment (martinholters) [inline test/misc.jl]: This now requires a `using LinearAlgebra`.

**Pull request closed without merging**